src/App.tsx

```diff
@@ -15,61 +14,185 @@ const goodsFromServer = [
+      ? visibleGoods.reverse()
+      : visibleGoods;
+  };
+
+  startedSort = () => (
+    this.setState({ isStarted: true })
+  );
+
+  sortByLength = () => (
+    this.setState((state) => ({
+      sortType: SortType.LENGTH,
+      isSortedByLength: !state.isSortedByLength,
+    }))
+  );
+
+  sortAlphabetically = () => (
+    this.setState((state) => ({
+      sortType: SortType.ALPHABET,
+      isSortedAlphabetically: !state.isSortedAlphabetically,
+    }))
+  );
+
+  revers = () => (
+    this.setState(state => ({ isReversed: !state.isReversed }))
+  );
+
+  handleChangeSelect = (event: ChangeEvent<HTMLSelectElement>) => (
+    this.setState({ selected: +event.target.value })
+  );
+
+  reset = () => (
+    this.setState(
+      { sortType: SortType.NONE, isReversed: false, selected: 1 },
+    )
+  );
+
+  render() {
+    const {
+      isStarted,
+      isReversed,
+      sortType,
+      selected,
+      isSortedByLength,
+      isSortedAlphabetically,
+    } = this.state;
+    const goods = this.getReorderedGoods(
+      goodsFromServer,
+      sortType,
+      isReversed,
+      selected,
+      isSortedByLength,
+      isSortedAlphabetically,
+    );
+
+    return (
+      <div className="App is-light">
+        {!isStarted && (
+          <button
+            type="button"
+            className="button is-success button-start"
+            onClick={this.startedSort}
+          >
+            Start
+          </button>
+        )}
+
+        {isStarted && (
+          <>
+            <div className="buttons">
+              <button
+                type="button"
+                className="button is-dark"
+                onClick={this.sortAlphabetically}
+              >
+                Sort alphabetically
+              </button>
+
+              <button
+                type="button"
+                className="button is-dark"
+                onClick={this.sortByLength}
+              >
+                Sort by length
+              </button>
+
+              <button
+                type="button"
+                className="button is-dark"
+                onClick={this.revers}
+              >
+                Reverse
+              </button>
+
+              <button
+                type="button"
+                className="button is-danger button-reset"
+                onClick={this.reset}
+              >
+                Reset
+              </button>
+
+              <select
+                name="minLength"
+                className="select is-primary"
+                value={selected}
+                onChange={this.handleChangeSelect}
+              >
+                {minLengths.map(length => (
+                  <option value={length} key={length}>
+                    {length}
+                  </option>
+                ))}
+              </select>
+            </div>
+
+            <ul className="Goods">
+              {goods.map(good => (
+                <li className="Goods__item" key={good}>
```

Comment:
What if we would have duplicated goods in the array ?

Reply:
I know, I was just too lazy to download the library key generator